src/Time.svelte

```diff
@@ -1,67 +1,74 @@
 <script>
   // @ts-check
 
-  /**
-   * Original timestamp
-   * @type {import("dayjs").ConfigType}
-   */
-  export let timestamp = new Date().toISOString();
+  const {
+    /**
+     * Original timestamp
+     * @type {import("dayjs").ConfigType}
+     */
+    timestamp = new Date().toISOString(),
 
-  /**
-   * Timestamp format for display.
-   * It's also used as a title in the `relative` mode
-   * @type {import("dayjs").OptionType}
-   * @example "YYYY-MM-DD"
-   */
-  export let format = "MMM DD, YYYY";
+    /**
+     * Timestamp format for display.
+     * It's also used as a title in the `relative` mode
+     * @type {import("dayjs").OptionType}
+     * @example "YYYY-MM-DD"
+     */
+    format = "MMM DD, YYYY",
 
-  /**
-   * Set to `true` to display the relative time from the provided `timestamp`.
-   * The value is displayed in a human-readable, relative format (e.g., "4 days ago", "Last week")
-   * @type {boolean}
-   */
-  export let relative = false;
+    /**
+     * Set to `true` to display the relative time from the provided `timestamp`.
+     * The value is displayed in a human-readable, relative format (e.g., "4 days ago", "Last week")
+     * @type {boolean}
+     */
+    relative = false,
 
-  /**
-   * Set to `true` to update the relative time at 60 second interval.
-   * Pass in a number (ms) to specify the interval length
-   * @type {boolean | number}
-   */
-  export let live = false;
-
-  /**
-   * Formatted timestamp.
-   * Result of invoking `dayjs().format()`
-   * @type {string}
-   */
-  export let formatted = "";
+    /**
+     * Set to `true` to update the relative time at 60 second interval.
+     * Pass in a number (ms) to specify the interval length
+     * @type {boolean | number}
+     */
+    live = false,
+    ...rest
+  } = $props();
 
   import { dayjs } from "./dayjs";
   import { onMount } from "svelte";
 
   /** @type {undefined | NodeJS.Timeout} */
   let interval = undefined;
+  let liveUpdate = $state(0);
 
   const DEFAULT_INTERVAL = 60 * 1_000;
 
   onMount(() => {
```

Comment:
Actually would be better to move it into $effect

```
$effect(() => {
		// This will be recreated whenever `milliseconds` changes
		const interval = setInterval(() => {
			count += 1;
		}, milliseconds);

		return () => {
			// if a callback is provided, it will run
			// a) immediately before the effect re-runs
			// b) when the component is destroyed
			clearInterval(interval);
		};
	});
	```

Reply:
good catch. that should be easy. i'll have a look tomorrow.

Reply:
i have updated the code to return the `clearInterval` in `$effect` return statement.